### DISCUSSION TITLE
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path

### DISCUSSION THREAD

**Discussion by blvckrv1n** [Homebrew/discussions#3160] [Getting started]
Guys I need help. I'll admit i'm a total beginner and hands down one of the biggest newbies. Every time I try to install homebrew it doesn't work and says something about fatal error... it gives me this.

HEAD is now at c65bb5d12 Merge pull request #13094 from Homebrew/org-discussions
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

But I've run 'brew --version' and it gives me

'Homebrew 3.4.5-29-gc65bb5d
Homebrew/homebrew-core (no Git repository)
Homebrew/homebrew-cask (git revision eb7d39ad37; last commit 2022-04-04)

can anyone help explain what i need to do or fix? thanks

**Comment by carlocab** [comment]
What is the output of `brew config` and `brew doctor`?

**Reply by blvckrv1n** [reply]
brew config gave me Last login: Thu Apr  7 01:02:57 on ttys002
hunter@hunters-Air ~ % brew config
HOMEBREW_VERSION: 3.4.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 33709297cf9cb37605cf4efcb7151905baf8ccf5
Last commit: 5 days ago
Core tap: N/A
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 13.0.0 build 1300
Git: 2.15.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6.5-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A

brew doctor gave me nothing
error: no available tap homebrew/core.

**Reply by carlocab** [reply]
Try doing `brew tap homebrew/core`.

**Reply by blvckrv1n** [reply]
sweet i think that worked. it finished installing everything , it says tapped 3 commands and 6028 formulae (6,366 files, 508.5mb) 
is that everything basic? am i going to need to install it again?

**Reply by carlocab** [reply]
What is the output of `brew config` and `brew doctor` now?

**Reply by blvckrv1n** [reply]
it tells me installation successful so thank you i've never got that far before haha. xD 
but below that it says next steps: run brew help to get started. so now i need a formula and cask but idk what to do. I was mainly trying to use it to mess with some old consoles i own but idk what i need to do for that.. just know videos i'm watching say i need homebrew lol

**Reply by Catalyst4Change** [reply]
hey @carlocab I have the same Issue I'm hoping you can help me as well :p 
I was getting the same response as  @blvckrv1n while trying to install homebrew. I added in `brew tap homebrew/core` and that changed the response, but I still don't think it's working. I can do `brew config` and it looks normal. when I do `brew doctor`, it gives me a huge set of file paths followed by `Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:  git -C $(brew --repo homebrew/core) checkout`. If I type THAT in, it gives me `fatal: You are on a branch yet to be born`, which honestly sounds like a threat. wtf is going on here and how can I right this ship? thanks!

**Reply by gromgit** [reply]
A rule of thumb when asking for technical help in three words: *Show, don't tell.* You've described the problem in your own words, i.e. "_told_ a story", but what folks trying to help really need are the exact details of errors and stuff, i.e. "_show_ us what actually happened".

Specifically, as @carlocab asked previously, what's _your_ output of `brew config` and `brew doctor`?

**Reply by Catalyst4Change** [reply]
cool, thank you. that's good advice.  If it wasn't obvious, I'm new at this. 

brew config:
`HOMEBREW_VERSION: 3.4.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 09e766f83c8a420cce9e2210c46a24ec3864a094
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 13.1.6 build 1316
Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3.1-x86_64
CLT: 13.3.1.0.1.1648687083
Xcode: N/A
Rosetta 2: true`

brew doctor:
`Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew/homebrew-core was not tapped properly! Run:
  rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10-config
  /Library/Frameworks/Python.framework/Versions/3.10/bin/python3-config

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
...
[hundreds of file paths]
...

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master`

**Reply by gromgit** [reply]
From your `brew config` output:
>```
> Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
> Core tap HEAD: (none)
> Core tap last commit: never
> Core tap branch: HEAD
>```

This is definitely not "normal": your core tap effectively doesn't exist. To fix:
>```
> rm -fr $(brew --repo homebrew/core)
> brew tap homebrew/core
>```

**Reply by Catalyst4Change** [reply]
Ok, thanks! that definitely changed things. alright, so now I have the following responses:

`brew config`

> `HOMEBREW_VERSION: 3.4.7
> ORIGIN: https://github.com/Homebrew/brew
> HEAD: 09e766f83c8a420cce9e2210c46a24ec3864a094
> Last commit: 2 days ago
> Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
> Core tap HEAD: bf33fb5785c4e99114ecde04f615f38d3da30991
> Core tap last commit: 27 minutes ago
> Core tap branch: master
> HOMEBREW_PREFIX: /usr/local
> HOMEBREW_CASK_OPTS: []
> HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
> HOMEBREW_MAKE_JOBS: 8
> Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
> CPU: octa-core 64-bit westmere
> Clang: 13.1.6 build 1316
> Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
> Curl: 7.79.1 => /usr/bin/curl
> macOS: 12.3.1-x86_64
> CLT: 13.3.1.0.1.1648687083
> Xcode: N/A
> Rosetta 2: true
> `

`brew doctor`

>  `Please note that these warnings are just used to help the Homebrew maintainers
> with debugging if you file an issue. If everything you use Homebrew for is
> working fine: please don't worry or file an issue; just ignore this. Thanks!
> 
> Warning: "config" scripts exist outside your system or Homebrew directories.
> `./configure` scripts often look for *-config scripts to determine if
> software packages are installed, and which additional flags to use when
> compiling and linking.
> 
> Having additional scripts in your path can confuse software installed via
> Homebrew if the config script overrides a system or Homebrew-provided
> script of the same name. We found the following "config" scripts:
>   /Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10-config
>   /Library/Frameworks/Python.framework/Versions/3.10/bin/python3-config
> 
> Warning: Unbrewed header files were found in /usr/local/include.
> If you didn't put them there on purpose they could cause problems when
> building Homebrew formulae, and may need to be deleted.
> 
> Unexpected header files:
>   /usr/local/include/node/cppgc/common.h
>  /usr/local/include/node/js_native_api.h
>   /usr/local/include/node/js_native_api_types.h
>   /usr/local/include/node/libplatform/libplatform-export.h
>   /usr/local/include/node/libplatform/libplatform.h
>   /usr/local/include/node/libplatform/v8-tracing.h
>  [...the files go on forever]
> `

**Reply by gromgit** [reply]
Your `HOMEBREW_PREFIX` is the standard one (`/usr/local`), so all your installs should be from precompiled bottles. You can therefore probably ignore all the `brew doctor` warnings that involve _building_ Homebrew formulae.

**Reply by Catalyst4Change** [reply]
thanks! everything _seems_ in order. Appreciate your help!
